packages/blade/src/utils/makeAccessible/makeAccessible.native.ts

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable guard-for-in */
+/* eslint-disable @typescript-eslint/no-unsafe-argument */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {
+  accessibilityMap,
+  accessibilityStateKeys,
+  accessibilityValueKeys,
+  supportedAccessibilityRoles,
+} from './accessibilityMapNative';
+import type { AccessibilityMap, AccessibilityProps } from './makeAccessible.d';
+import webToAccessibilityRole from './webToNativeRole';
+
+function isAccessibilityStateProp(prop: string): boolean {
+  return accessibilityStateKeys.includes(prop);
+}
+
+function isAccessibilityValueProp(prop: string): boolean {
+  return accessibilityValueKeys.includes(prop);
+}
+
+const makeAccessible = (props: Partial<AccessibilityProps>): Record<string, unknown> => {
+  const newProps: Record<string, any> = {};
+
+  // loop through all the incoming props and map them
+  for (const key in props) {
+    const propKey = key as keyof AccessibilityMap;
+    const propValue = props[propKey];
+    const accessibilityAttribute = accessibilityMap[propKey];
+
+    // group accesibilityState prop for native
+    if (isAccessibilityStateProp(propKey)) {
+      newProps.accessibilityState = {
+        ...newProps.accessibilityState,
+        [accessibilityAttribute]: propValue,
+      };
+      continue;
+    }
+
+    // group accesibilityValue prop for native
+    if (isAccessibilityValueProp(propKey)) {
+      newProps.accessibilityValue = {
+        ...newProps.accessibilityValue,
+        [accessibilityAttribute]: propValue,
+      };
+      continue;
+    }
+
+    if (accessibilityAttribute) {
+      newProps[accessibilityAttribute] = propValue;
+    } else {
+      console.warn(
+        `[Blade: makeAccessible]: No mapping found for ${propKey}. Make sure you have entered valid key`,
+      );
+    }
+  }
+
+  if (newProps.accessibilityRole) {
+    // map web to native overlapping roles
+    const role = webToAccessibilityRole(newProps.accessibilityRole);
+    newProps.accessibilityRole = role;
```

Comment:
can you share an example? didn't quite understand this?

Reply:
There are common overlaps between some of the `roles` between web and native. for example in web `slider` maps to `adjustable` in native. 

Example: 
https://github.com/razorpay/blade/pull/481/files#diff-b10f9269d534b1f77c6207692521ca38f0790ea3da3d0d45252deb29d1a8f81dR18